src/main/java/Duke.java

```diff
@@ -1,10 +1,275 @@
+
+    private static ArrayList<Task> list = new ArrayList<Task>();
+
+    public static void initList() {
+        try {
+            File file = new File(FILEPATH);
+            Scanner s = new Scanner(file); // create a Scanner using the File as the source
+            while (s.hasNext()) {
+                String input = s.nextLine();
+                String[] inputWords = input.split(" / ");
+                switch(inputWords[0]) {
+                case "T":
+                    list.add(new Todo(inputWords[2]));
+                    if (inputWords[1].equals("1")) {
+                        list.get(list.size()-1).markAsDone();
+                    }
+                    break;
+                case "D":
+                    list.add(new Deadline(inputWords[2], inputWords[3]));
+                    if (inputWords[1].equals("1")) {
+                        list.get(list.size()-1).markAsDone();
+                    }
+                    break;
+                case "E":
+                    list.add(new Event(inputWords[2], inputWords[3]));
+                    if (inputWords[1].equals("1")) {
+                        list.get(list.size()-1).markAsDone();
+                    }
+                    break;
+                default:
+                    throw new DukeException();
+                }
+            }
+            s.close();
+        } catch (FileNotFoundException e) {
+            createFolder();
+            createFile();
+        } catch (DukeException | ArrayIndexOutOfBoundsException e) {
+            System.out.println(LINE + "\nInput format within file is wrong!\n" + LINE);
+        }
+    }
+
+    public static void createFolder() {
+        File folder = new File(FOLDERPATH);
+        boolean folderIsCreated = folder.mkdir();
+        if (folderIsCreated) {
+            System.out.println("New folder created for storage\n" + LINE);
+        }
+    }
+
+    public static void createFile() {
+        File file = new File(FILEPATH);
+        try {
+            boolean fileIsCreated = file.createNewFile();
+            if (fileIsCreated) {
+                System.out.println("New text file created for storage\n" + LINE);
+            }
+        } catch (IOException e) {
+            System.out.println("Something went wrong: " + e.getMessage());
+        }
+    }
+
+    private static void appendToFile(String textToAppend) throws IOException {
+        File file = new File(FILEPATH);
+        if (file.length() != 0) {
+            textToAppend = System.lineSeparator() + textToAppend;
+        }
+        FileWriter fw = new FileWriter(FILEPATH, true); // create a FileWriter in append mode
+        fw.write(textToAppend);
+        fw.close();
+    }
+
+    private static void updateFile() {
+        try {
+            clearFile();
+            for (Task task : list) {
+                String textToAppend = "";
+                if (task instanceof Todo) {
+                    String isDone = task.isDone() ? "1" : "0";
```

Comment:
isDone sounds like a name developers would normally give to booleans. Perhaps you may want to use names such as completionStatus so that it is more clear that this variable is a string.